tests/testsuite/env.rs

```diff
@@ -721,3 +723,49 @@ fn test_parse_uint_default() {
     let config: TestUint = config.try_deserialize().unwrap();
     assert_eq!(config.int_val, 42);
 }
+
+#[test]
+fn test_non_utf8() {
```

Comment:
Would you be willing to split this into multiple commits
1. Add the test with a `should_panic` so it passes
2. Change behavior and remove `should_panic` so it passes

In cargo, we've found structuring PRs like this to be a big help
- tests the tests (was that actually invalid utf8 data?)
- Makes clear the intent of the change to reviewers and the community (less important here as its more obvious)